### PR TITLE
PEO-317 Update structure

### PR DESCRIPTION
**Pull request by chubchenko**
[PEO-317](https://garage.atlassian.net/browse/PEO-317)

### Description

- Extract logic from `Runner` to `Plugin`
- Update CLI and Collector
- Update Version representation
- Update way of configuration
- Update Logger

### Before submitting the merge request make sure the following are checked

- [x] Followed the style guidelines of this project
- [x] Performed a self-review of own code
- [x] Wrote the tests that prove that fix is effective/that feature works

### CODE REVIEW

File: lib/inquisition/active_record_doctor/runner.rb [hunk 1]
```diff
@@ -30,6 +30,10 @@ def create_message(ard_task, issue_object, details)
         issue_text = ard_task.to_s.split('::').last.split(/(?=[A-Z])/).map(&:downcase).join(' ')
         "#{issue_object} has #{issue_text}, details: #{details ? details.join(', ') : 'n/a'}"
       end
+
+      def load_environment
```
Comment (D3N) [inline lib/inquisition/active_record_doctor/runner.rb:34]: it's not DRY

File: spec/inquisition/plugin_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    context 'when there are no descendants' do
+      it { expect(dummy.collection).to be_empty }
+    end
+
+    context 'when there is at least one descendant' do
+      let(:descendant) do
+        Class.new(dummy)
+      end
+
+      it do
```
Comment (D3N) [inline spec/inquisition/plugin_spec.rb]: it's hard to read this example, i would make it
```expect { descendant }.to change(dummy.collection, :size).by(1)```

**Review by D3N — APPROVED** [review]

**Review by rostikkkk2 — APPROVED** [review]

File: spec/inquisition/plugin_spec.rb (hunk 2 above)
Comment (chubchenko) [reply spec/inquisition/plugin_spec.rb]: Fixed https://github.com/rubygarage/inquisition/pull/74/commits/8832c1308cc49c1976bfe623eb0d3aea110ce53c#diff-475fb638797e13b4dcc5149eeb7cb8cdR18

File: lib/inquisition/active_record_doctor/runner.rb (hunk 1 above)
Comment (chubchenko) [reply lib/inquisition/active_record_doctor/runner.rb:34]: Yes, you are right, it will be fixed in the next iteration.

**Pull request merged**